src/Pomelo.EntityFrameworkCore.MySql/Query/ExpressionTranslators/Internal/MySqlDatePartTranslator.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Pomelo Foundation. All rights reserved.
+// Licensed under the MIT. See LICENSE in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Query.Expressions;
+using Microsoft.EntityFrameworkCore.Query.Expressions.Internal;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.EntityFrameworkCore.Query.ExpressionTranslators.Internal
+{
+    public class MySqlDatePartTranslator : IMemberTranslator
+    {
+        public virtual Expression Translate([NotNull] MemberExpression memberExpression)
+        {
+            string datePart;
+            if (memberExpression.Expression.Type == typeof(DateTime)
+                && (datePart = GetDatePart(memberExpression.Member.Name)) != null)
+            {
+                return new DatePartExpression(datePart, memberExpression.Type, memberExpression.Expression);
+            }
+            return null;
+        }
+
+        private static string GetDatePart(string memberName)

```

Comment:
The GetDatePart is subtlety different between classes on the case statements:

`case nameof(DateTime.AddYears):` -vs- `case nameof(DateTime.Year):`

Please let me know if I misunderstood your question.  Thanks.